src/arithmetic/modular_set.cc

```diff
@@ -32,10 +32,27 @@ TVM_STATIC_IR_FUNCTOR(IRPrinter, vtable)
 
 
 // internal entry for const int bound
+// This condition holds for all instances: coeff >= 0, base in [0, coeff]
 struct ModularSetAnalyzer::Entry {
   int64_t coeff{1};
   int64_t base{0};
 
+  Entry() = default;
+
+  Entry(int64_t coeff, int64_t base) {
+    this->coeff = coeff;
+
+    if (coeff < 0) {
+      coeff = -coeff;
+    }
+
+    if (coeff != 0) {
```

Comment:
> C++ constructor also have a restriction of not throw exception

As far as I know, throwing exceptions from constructors is absolutely OK. (Unless you want to avoid exceptions completely and return an error code or an `optional`, in which case you have to use a factory function.)